source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -415,6 +450,60 @@ void ProcessorState::continueIfNecessary() {
   }
 }
 
+bool ProcessorState::handleStreamedBodyResponse(const CommonResponse& common_response) {
+  Buffer::OwnedImpl chunk_data;
+  auto chunk = dequeueStreamingChunk(chunk_data);
+  ENVOY_BUG(chunk != nullptr, "Bad streamed body callback state");
+  if (common_response.has_body_mutation()) {
+    ENVOY_LOG(debug, "Applying body response to chunk of data. Size = {}", chunk->length);
+    MutationUtils::applyBodyMutations(common_response.body_mutation(), chunk_data);
+  }
+  bool should_continue = chunk->end_stream;
+  if (chunk_data.length() > 0) {
+    ENVOY_LOG(trace, "Injecting {} bytes of data to filter stream", chunk_data.length());
+    injectDataToFilterChain(chunk_data, chunk->end_stream);
+  }
+
+  if (queueBelowLowLimit()) {
+    clearWatermark();
+  }
+  if (chunk_queue_.empty()) {
+    onFinishProcessorCall(Grpc::Status::Ok);
+  } else {
+    onFinishProcessorCall(Grpc::Status::Ok, callback_state_);
+  }
+
+  return should_continue;
+}
+
+bool ProcessorState::handleDuplexStreamedBodyResponse(const CommonResponse& common_response) {
+  const auto& streamed_response = common_response.body_mutation().streamed_response();
+  const auto& body = streamed_response.body();
+  const bool end_of_stream = streamed_response.end_of_stream();
+
+  if (body.size() > 0) {
+    Buffer::OwnedImpl buffer;
+    buffer.add(body);
+    ENVOY_LOG(trace,
+              "Injecting {} bytes of data to filter stream in FULL_DUPLEX_STREAMED mode. "
+              "end_of_stream is {}",
+              buffer.length(), end_of_stream);
+    injectDataToFilterChain(buffer, end_of_stream);
+  }
+
+  if (end_of_stream) {
+    onFinishProcessorCall(Grpc::Status::Ok);
+  } else {
+    // Set the state to CallbackState::StreamedBodyCallback to wait for more bodies.
+    // However, this could be the last chunk of body, and trailers are right after it.
+    // The function to handle trailers response needs to consider this.
+    onFinishProcessorCall(Grpc::Status::Ok, CallbackState::StreamedBodyCallback);
+  }
+
+  const bool should_continue = end_of_stream;
+  return should_continue;
```

Comment:
I think here you don't need this local variable and assignment. You can just return  `end_of_stream`

You can have code comment explain  end_of_stream = should_continue

Reply:
done